packages/ics721/src/execute.rs

```diff
@@ -299,13 +317,24 @@ where
 
     /// Default implementation using `cw721_base::msg::InstantiateMsg`
     fn init_msg(&self, _deps: Deps, env: &Env, class: &Class) -> StdResult<Binary> {
-        to_binary(&cw721_base::msg::InstantiateMsg {
-            // Name of the collection MUST be class_id as this is how
-            // we create a map entry on reply.
+        // use by default ClassId, in case there's no class data with name and symbol
+        let mut instantiate_msg = cw721_base::msg::InstantiateMsg {
             name: class.id.clone().into(),
             symbol: class.id.clone().into(),
             minter: env.contract.address.to_string(),
-        })
+        };
+
+        // unwrapped to collection data and in case of success, set name and symbol
+        if let Some(binary) = class.data.clone() {
+            let class_data_result: StdResult<CollectionData> = from_json(binary);
```

Comment:
part 2, instantiate2: using symbol and name, in case class data holds collection data

Reply:
Who is calling this init_msg function and what Class object are they passing to it?